### ISSUE TITLE
number_item comparison error when null

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#232]
```
2021-03-14 11:07:39.950 [ERROR] [jsr223.jruby.x     ] - undefined method `to_big_decimal' for nil:NilClass
Did you mean?  BigDecimal (NoMethodError)
/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-3.2.1/lib/openhab/dsl/items/number_item.rb:196:in `numeric_compare'
/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-3.2.1/lib/openhab/dsl/items/number_item.rb:82:in `<=>'
```

**Comment by boc-tothefuture** [comment]
What was the context this was used in?

**Comment by jimtng** [comment]
I don't remember the exact code now, but it would've been something like `if Item1 > 10`, the error would happen when the state is nil. It's similar to other errors. I guess it caught me by surprise still. Perhaps this is classified as user error and the exception would stop the code, which is what should happen anyway?

**Comment by ccutrer** [comment]
yes, this should be considered user error (see https://github.com/boc-tothefuture/openhab-jruby/issues/223#issuecomment-944846479). this particular case I expect would raise an ArgumentError `comparison of Java::OrgOpenhabCoreLibraryItems::NumberItem with Integer failed`.

**Comment by jimtng** [comment]
IMO:

```
rule 'x' do
  on_start
  run do
    # logger.error('MORE') if Number1 > 10 # This should raise an error
    # logger.error('LESS') if Number1 < 10 # This should raise an error
    logger.error('Equal') if Number1 == 10 # This should NOT raise an error, simply because NULL is not the same as 10
    logger.error('Equal') if Number1 == Number2 # This should NOT raise an error because NULL == NULL
  end
end
```

**Comment by jimtng** [comment]
Fixed in #299
